packages/modules/src/modules/banners/climateMoment3D/components/DraggableOverlay.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useRef, useState } from 'react';
+import { Canvas, useThree, useFrame } from '@react-three/fiber';
+import { a } from '@react-spring/three';
+import { useDrag } from 'react-use-gesture';
+import THREE from 'three';
+import { SerializedStyles, css } from '@emotion/react';
+
+const canvasContainer = css`
+    width: 100%;
+    height: 100%;
+`;
+
+function DraggableMesh() {
+  const [position, setPosition] = useState<[x: number, y: number, z: number]>([0, 0, 0]);
+  const ref = useRef<THREE.Mesh>(null);
+
+  const { size, viewport } = useThree();
+  const aspect = size.width / viewport.width;
+
+  useFrame(() => {
+    if(ref.current) {
+      ref.current.rotation.z += 0.005;
+    }
+  });
+
+  const bind = useDrag(
+    ({ offset: [x, y] }) => {
+      const [,, z] = position;
+      setPosition([x / aspect, -y / aspect, z]);
+    },
+    { pointerEvents: true }
+  );
+
+  return (
+    <a.mesh
```

Comment:
There is an event handler typing mismatch between the resultant type of the `bind` var and the expected type for the `a.mesh` i.e. three animated mesh type